### PR TITLE
PLT-3412 WebRTC Server side & System Console

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This is the server side implementation (configs, tokens, etc) for WebRTC with the correspondant System Console Settings

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3412

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs)
- [x] All new/modified APIs include changes to the drivers
- [x] Has enterprise changes (https://github.com/mattermost/enterprise/pull/49)
- [x] Has UI changes
- [x] Includes text changes and localization files updated
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by enahum** [comment]
it seems to be failing due to an OAuth test that depends on an Enterprise feauture not merged

**Comment by jasonblais** [comment]
Added WIP label until [Asaad's UI tweaks have been included](https://mattermost.atlassian.net/browse/PLT-2818)

**Comment by enahum** [comment]
@jasonblais @asaadmahmood tweaks are going to another PR, this one is only server side

**Comment by jasonblais** [comment]
Cool, will remove PM tag and add it to the upcoming PR

**Comment by coreyhulen** [comment]
+1

File: model/config.go [hunk 1]
```diff
@@ -1130,3 +1148,66 @@ func (o *Config) Sanitize() {
...
+
+func (o *Config) isValidWebrtcSettings() *AppError {
+	if *o.WebrtcSettings.Enable {
+		if len(*o.WebrtcSettings.GatewayWebsocketUrl) == 0 || !IsValidWebsocketUrl(*o.WebrtcSettings.GatewayWebsocketUrl) {
+			return NewLocAppError("Config.IsValid", "model.config.is_valid.webrtc_gateway_ws_url.app_error", nil, "")
+		} else if len(*o.WebrtcSettings.GatewayAdminUrl) == 0 || !IsValidHttpUrl(*o.WebrtcSettings.GatewayAdminUrl) {
+			return NewLocAppError("Config.IsValid", "model.config.is_valid.webrtc_gateway_admin_url.app_error", nil, "")
+		} else if len(*o.WebrtcSettings.GatewayAdminSecret) == 0 {
+			return NewLocAppError("Config.IsValid", "model.config.is_valid.webrtc_gateway_admin_secret.app_error", nil, "")
+		} else if len(*o.WebrtcSettings.TurnURI) != 0 {
```
Comment (hmhealey) [inline model/config.go]: Should you be checking if the TurnURI is a valid URL here?

**Comment by hmhealey** [comment]
Also, this isn't particularly important, but can you try to stay consistent with capitalization of WebRTC? You're referring to it as `Webrtc`, `WebRtc`, and `WebRTC` in different places which seems like it's going to be a nightmare when working on it without a proper autocomplete. That's more of a personal thing though so feel free to leave it if no one else has an issue with it.

**Comment by enahum** [comment]
yeah sure, I'll change the consistency of the `webrtc` word everywhere

File: model/config.go (hunk 1 above)
Comment (enahum) [inline model/config.go]: regarding this one the turn and stun uri are like turn:address:port and stun:address:port, I'll create a function to validate them.

**Comment by enahum** [comment]
@hmhealey done with the changes

**Comment by coreyhulen** [comment]
I'm re +1 just in case.

**Comment by asaadmahmood** [comment]
```
asdasd

**Pull request merged**